tests/lib/mkpinentry.sh

```diff
@@ -1,6 +1,18 @@
 #!/bin/sh
 
-echo "setup fake gpg pinentry environment"
-mkdir -p ~/.snap/gnupg/
-echo pinentry-program "$TESTSLIB/pinentry-fake.sh" > ~/.snap/gnupg/gpg-agent.conf
-chmod -R go-rwx ~/.snap
+if [ gpg --version | MATCH "gpg (GnuPG) 1." ]; then
```

Comment:
umm... how does this work? `MATCH` is `grep -E`, and this means that `(` and `)` are metacharacters, so `gpg (GnuPG)` will not match `gpg (GnuPG)`:
```sh-session
$ gpg --version | grep -E 'gpg (GnuPG) 1.'; echo $?
1
```